File: ros2/src/fake_depth_sensor/fake_depth_sensor/depth_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float64
import math

class DepthPublisherNode(Node):
    def __init__(self):
        super().__init__('depth_publisher')
        self.publisher_ = self.create_publisher(Float64, 'depth', 10)
        self.timer_ = self.create_timer(0.5, self.publish_depth)
        self.time_x_ = 0.0
        self.get_logger().info('Depth Publisher Node has been started.')

    def publish_depth(self):
        msg = Float64()
        msg.data = 50.0 + 50.0 * math.sin(self.time_x_)
        self.time_x_ += 0.2
        
        self.get_logger().info(f'Publishing Depth: {msg.data:.2f}')
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = DepthPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: rashwan/tasks/task 4/fake_depth_project/rov/src/fake_depth_sensor/src/depth_publisher.cpp
#include "fake_depth_sensor/depth_publisher.hpp"

#include <cmath>

DepthPublisher::DepthPublisher()
    : Node("depth_publisher"),
      time_(0.0)
{
    publisher_ = this->create_publisher<std_msgs::msg::Float64>(
        "/depth",
        10
    );

    timer_ = this->create_wall_timer(
        std::chrono::milliseconds(500),
        std::bind(&DepthPublisher::publish_depth, this)
    );

    RCLCPP_INFO(this->get_logger(), "Depth Publisher started.");
}

void DepthPublisher::publish_depth()
{
    std_msgs::msg::Float64 message;

    message.data = 5.0 + 4.0 * std::sin(time_);

    publisher_->publish(message);

    RCLCPP_INFO(
        this->get_logger(),
        "Published depth: %.2f m",
        message.data
    );

    time_ += 0.3;
}
